### PR TITLE
Reduce test timeout to 10mins and modify tests to fail before timeout

### PR DESCRIPTION
**Pull request by thevindu-w**

### CODE REVIEW

File: test-docker.sh [hunk 1]
```diff
@@ -28,7 +28,7 @@ while ! nc -zvn 127.0.0.1 7777 &>/dev/null; do
     sleep .2
 done
 
-timeout 1800 python3 -u tests/integration/test.py |& tee "logs/${run_id}_test.txt"
+timeout 600 python3 -u tests/integration/test.py |& tee "logs/${run_id}_test.txt"
```
Comment (miyurud) [inline test-docker.sh:31]: Can we move this value to a constant like `TIMEOUT_SECONDS` and use that here?

File: Dockerfile [hunk 2]
```diff
@@ -34,13 +34,24 @@ RUN make install
...
-COPY . .
+COPY ./GraphSAGE ./GraphSAGE
 RUN pip install -r ./GraphSAGE/requirements
 
+COPY ./conf ./conf
+COPY ./src ./src
+COPY ./src_python ./src_python
+COPY ./build.sh ./build.sh
+COPY ./CMakeLists.txt ./CMakeLists.txt
+COPY ./main.cpp ./main.cpp
```
Comment (miyurud) [inline Dockerfile:51]: Why do we need to copy C++ sources to docker image? Can't we just copy the executable and library dependencies?

Comment (thevindu-w) [reply Dockerfile:51]: We need C++ sources as they are compiled during Docker build

**Review by miyurud — CHANGES_REQUESTED** [review]
Please address the issues pointed in the PR.

File: test-docker.sh (hunk 1 above)
Comment (thevindu-w) [reply test-docker.sh]: Fixed that by creating a variable `TIMEOUT_SECONDS=600`

File: Dockerfile (hunk 2 above)
Comment (miyurud) [reply Dockerfile:51]: Ok

File: test-docker.sh (hunk 1 above)
Comment (miyurud) [reply test-docker.sh]: Ok

**Review by miyurud — APPROVED** [review]

**Pull request merged**